app/src/main/java/com/peluso/walletguru/model/Account.kt

```diff
@@ -6,22 +6,39 @@ import com.kirkbushman.araw.models.Submission
  * Class which represents a financial account to be tracked in our application. An account is linked
  * to certain subreddits, and based on the performance of the account these posts are prioritized
  */
-class Account(val type: AccountType, val currentBalance: Float) {
+class Account(val type: AccountType, val currentBalance: Float, val percentageChange: Float) {
 
     //TODO: make this able to be stored in a sqlite database or in sharedprefs so we can save a user's
     // balance over time
 
+
     companion object {
-        fun List<Account>.orderSubmissions(submissions: List<Submission>): List<Submission> {
-            //TODO: write the algorithm for how to order a list of submissions based on the
-            // performance of the account
-            return submissions
+        fun List<Account>.orderSubmissions(submissions: List<Submission>, accounts: List<Account>): List<Submission> {
+
+            //list of accounts sorted in order of decreasing percentage change
+            val sortedAccounts = accounts.sortedByDescending { it.percentageChange }
+
+            //map ranked accounts index to account type
+            val subredditRankings = sortedAccounts.map { it.type.toString() to sortedAccounts.indexOf(it) }.toMap()
```

Comment:
I'm not sure making the map key value a string with the account type will work. This is what the current values will look like when I tested it:
![image](https://user-images.githubusercontent.com/22043527/113427273-bbd1b580-93a2-11eb-8f7e-5f4d562083cb.png)
The submission object has no reference to the 'AccountType' which we've implemented after the API call goes in. We probably need to make this extension function for a `Map<AccountType, List<Submission>>`  so we can preserve which account type a certain posts belongs to